pkg/apis/networking/validation/validation.go

```diff
@@ -165,8 +165,10 @@ func ValidateIPBlock(ipb *networking.IPBlock, fldPath *field.Path) field.ErrorLi
 			allErrs = append(allErrs, field.Invalid(exceptPath, exceptIP, "not a valid CIDR"))
 			return allErrs
 		}
-		if !cidrIPNet.Contains(exceptCIDR.IP) {
-			allErrs = append(allErrs, field.Invalid(exceptPath, exceptCIDR.IP, "not within CIDR range"))
+		cidrMaskLen, _ := cidrIPNet.Mask.Size()
+		exceptMaskLen, _ := exceptCIDR.Mask.Size()
+		if !cidrIPNet.Contains(exceptCIDR.IP) || cidrMaskLen > exceptMaskLen {
```

Comment:
This allows except to be the same as the outer CIDR (10.0.0.0/24 except 10.0.0.0/24).  Is that a useful case?  Perhaps we should force it to be a longer mask if the CIDR IP and except IP are the same?  Or else we should add a testcase that it is semantically valid, but useless.

Reply:
Thanks @thockin for the suggestion, I agree it's not a useful case. Updated the condition and the message accordingly.